components/src/status_im/ui/components/react.cljs

```diff
@@ -144,18 +144,33 @@
             [text-class (dissoc options-with-style :nested?)]
             nested-text-elements)))
 
+(def text-input-refs (atom #{}))
+
 (defn text-input
   [options text]
-  [(text-input-class)
-   (merge
-    {:underline-color-android  :transparent
-     :max-font-size-multiplier max-font-size-multiplier
-     :placeholder-text-color   colors/text-gray
-     :placeholder              (i18n/label :t/type-a-message)
-     :value                    text}
-    (-> options
-        (update :style typography/get-style)
-        (update :style dissoc :line-height)))])
+  (let [input-ref (atom nil)]
+    (reagent/create-class
+      {:component-will-unmount #(when @input-ref
+                                  (swap! text-input-refs disj @input-ref))
```

Comment:
better to remove on ref nil , because it's not always unmounted

Reply:
You mean both in `component-will-unmount` and when ref is nil?

Reply:
At least for password input I noticed that it is not unmounted when navigating between accounts list/login screen, but it gets unmounted after login, must be due to react-navigation.

Reply:
i mean remove and add in ref function, if nil remove overwise add 
```
(if r 
 (swap! text-input-refs conj r) 
(when @input-ref
  (swap! text-input-refs disj @input-ref)
 (reset! input-ref nil)))
```

Reply:
@siphiuel wdyt about this ^

Reply:
Yes, agree with that, pushed a fix.

Reply:
so we don't need :component-will-unmount anymore then? we `disj` when ref is nil

Reply:
It seems that we still do. Here's a log excerpt after login:
08-21 15:40:20.975 26177 26244 I ReactNativeJS: INFO [status-im.ui.components.react:155] - # disj on unmount
08-21 15:40:20.977 26177 26244 I ReactNativeJS: INFO [status-im.ui.components.react:171] - # disj on ref nil
08-21 15:40:20.999 26177 26244 I ReactNativeJS: INFO [status-im.ui.components.react:168] - #conj on ref not nil
Actually, with this logic we will still have a dead ref in` text-input-refs`, because :ref callback is nil is always invoked before the one with the non-nil value.

Reply:
So maybe it makes sense to only remove on unmount, what do you think?

Reply:
the log shows we don't need it, because we have it in ref why do you want to `disj` twice the same object?

Reply:
The problem is that after `disj` (twice ) we `conj` it again, although it's not visible anymore. I am confused as to why we want that to happen if it's about to unmount (or already unmounted) anyway.

Reply:
disj twice is wrong, yes